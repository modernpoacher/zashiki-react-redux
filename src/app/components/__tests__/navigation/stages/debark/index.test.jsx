import React from 'react'
import PropTypes from 'prop-types'
import renderer from 'react-test-renderer'

import Debark from '#zashiki-react-redux/app/components/navigation/stages/debark'

jest.mock('#zashiki-react-redux/app/common/get-debark-route', () => jest.fn().mockReturnValue('MOCK DEBARK ROUTE'))

/**
 * @param {{ to: string | { pathname: string }, children: React.ReactNode | React.ReactNode[] }} param0
 * @returns {React.JSX.Element}
 */
function MockLink ({ to, children }) {
  if (typeof to === 'string') {
    return (
      <a href={to} className='mock-link'>
        {children}
      </a>
    )
  }

  const {
    pathname
  } = to

  return (
    <a href={pathname} className='mock-link'>
      {children}
    </a>
  )
}

MockLink.propTypes = {
  to: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.shape({})
  ]),
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(
      PropTypes.node
    )
  ])
}

jest.mock('react-router', () => {
  return {
    __esModule: true,
    Link: MockLink
  }
})

describe('#zashiki-react-redux/app/components/navigation/stages/debark', () => {
  describe('Always', () => {
    it('renders', () => {
      expect(renderer.create(<Debark />).toJSON())
        .toMatchSnapshot()
    })
  })
})
