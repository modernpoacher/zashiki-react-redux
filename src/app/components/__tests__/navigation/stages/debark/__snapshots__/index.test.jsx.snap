// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@modernpoacher/zashiki-react-redux/app/components/navigation/stages/debark Always renders 1`] = `
<a
  className="mock-link"
>
  Debark Stage
</a>
`;
